### ISSUE TITLE
Handle possible null HttpContext in ReplyForbiddenWithWwwAuthenticateHeaderAsync

### ISSUE THREAD

**Issue by pmaytak** [AzureAD/microsoft-identity-web#380]
Come up with a solution to handle possible null HttpContext in ReplyForbiddenWithWwwAuthenticateHeaderAsync in TokenAcquisition class.

https://github.com/AzureAD/microsoft-identity-web/blob/3c77e3c8f6fe1f8adad523b3278615a3585b8755/src/Microsoft.Identity.Web/TokenAcquisition.cs#L578-L582

**Comment by jmprieur** [comment]
We should have an extra parameter with a default value of null.
If passed-in we'd use it in the method, otherwise we'd use the HttpContextAccessor as today.

```CSharp
public async Task ReplyForbiddenWithWwwAuthenticateHeaderAsync(IEnumerable<string> scopes, MsalUiRequiredException msalServiceException, HttpContext httpContext = null)
```

Alternatively we could just pass-in the HttpResponse:
1. Pro this is all it needs
2. in term of usability this could be more complicated for customers to find they need to use HttpContext.Reponse ?

@jennyf19 @pmaytak what do you think?

**Comment by jmprieur** [comment]
@pmaytak
Let's go with providing the HttpResponse as a parameter

This will also fix Azure devops [Bug 1099665: Defect : CS8602, Component : src\Microsoft.Identity.Web\TokenAcquisition.cs (1 issue)](https://identitydivision.visualstudio.com/IDDP/_workitems/edit/1099665) and therefore assigning to you.

**Comment by jmprieur** [comment]
Updated the wiki:
https://github.com/AzureAD/microsoft-identity-web/wiki/Managing-incremental-consent-and-conditional-access#handling-incremental-consent-or-conditional-access-in-web-apis

cc: @pmaytak @jennyf19
